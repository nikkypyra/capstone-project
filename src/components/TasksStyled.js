import React from 'react'
import styled from 'styled-components/macro'
import Checkbox from './Checkbox'
import DeleteButton from './Buttons/DeleteButton'
import PropTypes from 'prop-types'
import { db } from '../firebase'

TasksStyled.propTypes = {
  pets: PropTypes.array,
  setPets: PropTypes.func,
}

export default function TasksStyled({ pet, tasks }) {
  const todos = tasks.filter(function (task) {
    return task.petId === pet.id
  })
  //const todos = tasks || []
  return (
    <TaskWrapper>
      {todos
        .slice()
        .sort(
          (taskA, taskB) =>
            taskA.date.localeCompare(taskB.date) ||
            taskA.time.localeCompare(taskB.time)
        )
        .map((todo) => (
          <section key={todo.id}>
            <div className="marker">
              <img
                src={process.env.PUBLIC_URL + '/images/taskpaw.png'}
                alt=""
              />
            </div>
            <div className="description">
              <h3>{todo.description}</h3>
            </div>
            <div className="time">
              <h4>{todo.time}</h4>
            </div>
            <div className="date">
              <p>{todo.date}</p>
            </div>
            <div className="person">
              <p>To be completed by: {todo.person}</p>
            </div>
            <div className="status">
              <Checkbox
                checked={todo.complete}
                onChange={() => handleCheckbox(todo)}
              ></Checkbox>
            </div>
            <div className="delete">
              <DeleteButton onClick={() => deleteTask(todo)} />
            </div>
          </section>
        ))}
    </TaskWrapper>
  )
  function handleCheckbox(todo) {
    db.collection('tasks').doc(todo.id).update({ complete: !todo.complete })
  }

  function deleteTask(todo) {
    db.collection('tasks').doc(todo.id).delete()
  }

  /* function handleCheckbox(task, id) {
    const index = pets.findIndex((pet) => task.petId === pet.id)
    const pet = pets[index]
    const petsTasks = pet.tasks
    const taskIndex = petsTasks.findIndex((task) => task.id === id)
    const updatedTask = tasks[taskIndex]
    const newTask = { ...updatedTask, complete: !updatedTask.complete }
    const newTaskList = [
      ...petsTasks.slice(0, taskIndex),
      { ...newTask },
      ...petsTasks.slice(taskIndex + 1),
    ]
    const updatedPet = { ...pet, tasks: newTaskList }
    setPets([
      ...pets.slice(0, index),
      { ...updatedPet },
      ...pets.slice(index + 1),
    ])
  }

  function deleteTask(task, id) {
    const index = pets.findIndex((pet) => task.petId === pet.id)
    const pet = pets[index]
    const petsTasks = pet.tasks
    const taskIndex = petsTasks.findIndex((task) => task.id === id)
    const newTaskList = [
      ...petsTasks.slice(0, taskIndex),
      ...petsTasks.slice(taskIndex + 1),
    ]
    const updatedPet = { ...pet, tasks: newTaskList }
    setPets([
      ...pets.slice(0, index),
      { ...updatedPet },
      ...pets.slice(index + 1),
    ])
  } */
}

const TaskWrapper = styled.main`
  section {
    margin: 16px 0px;
    display: grid;
    grid-template-columns: 1.4fr 2fr 3fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
    justify-content: space-evenly;
    align-items: center;
  }

  .marker {
    grid-row: 1/3;
    grid-column: 1/2;

    img {
      height: 40px;
      width: 40px;
      border: 4px solid var(--tertiary);
      border-radius: 50%;
    }
  }

  .description {
    grid-row: 1/2;
    grid-column: 2/5;
  }

  .time {
    grid-row: 2/3;
    grid-column: 2/3;
  }

  .delete {
    grid-row: 1/2;
    grid-column: 6/7;
  }
  .date {
    grid-row: 2/3;
    grid-column: 3/4;
  }

  .person {
    grid-row: 3/4;
    grid-column: 2/6;
  }

  .status {
    grid-row: 2/3;
    grid-column: 4/5;
  }
`
