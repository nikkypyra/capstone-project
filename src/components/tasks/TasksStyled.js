import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import Checkbox from './Checkbox'
import pawSrc from '../../images/taskpaw.png'

TasksStyled.propTypes = {
  todo: PropTypes.object,
  handleCheckbox: PropTypes.func.isRequired,
}

export default function TasksStyled({ todo, handleCheckbox }) {
  return (
    <>
      <Marker>
        <img src={pawSrc} alt="" />
      </Marker>
      <Description>
        <h3>{todo.description}</h3>
      </Description>
      <Time>
        <h4>{todo.time}</h4>
      </Time>
      <Date data-cy="date">
        <p>{todo.date}</p>
      </Date>
      <Person data-cy="owner_name">
        <p>To be completed by: {todo.person}</p>
      </Person>
      <Status>
        <Checkbox
          checked={todo.complete}
          onChange={() => handleCheckbox(todo)}
        ></Checkbox>
      </Status>
    </>
  )
}

const Marker = styled.div`
  grid-row: 1/3;
  grid-column: 1/2;

  img {
    height: 40px;
    width: 40px;
    border: 4px solid var(--tertiary);
    border-radius: 50%;
  }
`

const Description = styled.div`
  grid-row: 1/2;
  grid-column: 2/5;
`

const Time = styled.div`
  grid-row: 2/3;
  grid-column: 2/3;
`

const Date = styled.div`
  grid-row: 2/3;
  grid-column: 3/4;
`

const Person = styled.div`
  grid-row: 3/4;
  grid-column: 2/6;
`

const Status = styled.div`
  grid-row: 2/3;
  grid-column: 4/5;
`
