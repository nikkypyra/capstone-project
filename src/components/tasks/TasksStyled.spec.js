import React from 'react'
import { render, cleanup } from '@testing-library/react'
import TasksStyled from './TasksStyled'
import { MemoryRouter as Router } from 'react-router-dom'

afterEach(cleanup)

const taskMock = {
  description: 'Walk',
  date: '2020-06-01',
  time: '08:00',
  person: 'Martin',
  complete: true,
}

const handleCheckbox = jest.fn()

describe('<TasksStyled />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <Router>
        <TasksStyled todo={taskMock} handleCheckbox={handleCheckbox} />
      </Router>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
