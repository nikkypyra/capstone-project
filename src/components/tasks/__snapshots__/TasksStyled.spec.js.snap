// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksStyled /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="TasksStyled__Marker-j2i96a-0 kTppzS"
  >
    <img
      alt=""
      src="taskpaw.png"
    />
  </div>
  <div
    class="TasksStyled__Description-j2i96a-1 dAkghC"
  >
    <h3>
      Walk
    </h3>
  </div>
  <div
    class="TasksStyled__Time-j2i96a-2 DoAZa"
  >
    <h4>
      08:00
    </h4>
  </div>
  <div
    class="TasksStyled__Date-j2i96a-3 kHFcCQ"
    data-cy="date"
  >
    <p>
      2020-06-01
    </p>
  </div>
  <div
    class="TasksStyled__Person-j2i96a-4 jToTCz"
    data-cy="owner_name"
  >
    <p>
      To be completed by: Martin
    </p>
  </div>
  <div
    class="TasksStyled__Status-j2i96a-5 fCOCIY"
  >
    <input
      checked=""
      class="Checkbox__CheckboxStyled-sc-1blqffb-0 gefyIW"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
